src/nvim/tui/tui.c

```diff
@@ -131,7 +131,7 @@ static void terminfo_start(UI *ui)
   data->can_use_terminal_scroll = true;
   data->bufpos = 0;
   data->bufsize = sizeof(data->buf) - CNORM_COMMAND_MAX_SIZE;
-  data->showing_mode = 0;
+  data->showing_shape = SHAPE_IDX_N;
```

Comment:
Why not keep mode as the driver? It's more general and may be used to decide things other than shape.